scripts/combine_readme

```diff
@@ -0,0 +1,107 @@
+#!/bin/bash -e
+#
+# Copyright 2018-present The Material Components for iOS Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+usage() {
+  echo
+  echo "Usage: $0 <Topic> <path>"
+  echo
+  echo "Generates a root readme file by combining multiple readme files from a subfolder."
+  echo
+  echo "Example usage: $0 releasing contributing"
+}
+
+if [ "$#" -ne 2 ]; then
+  usage
+  exit 1
+fi
+
+if [ "$#" -eq 0 ]; then
+    echo "No arguments supplied"
+fi
+
+# Expand all files inline
+# Args: $1 - tempoaray output path (ie: #$TMP_README_PATH or TMP_TOPIC_PATH)
+#       $2 - topic doc path (ie: $COMPONENT_PATH/docs or $TOPIC_PATH)
+#				$3 - expanded topic path (ie: $TMP_EXPANDED_README_PATH or $TMP_EXPANDED_TOPIC_PATH)
+# Example Usage: expand_inline_references $TMP_TOPIC_PATH 
+expand_inline_references() {
+
+  tmp_output_path="$1"			# $TMP_TOPIC_PATH
+  topic_doc_path="$2" 			# $TOPIC_PATH
+  expanded_topic_path="$3"	# $TMP_EXPANDED_TOPIC_PATH
+	IFS='' # Don't trim whitespace from lines
+	cat "$tmp_output_path" | while read line; do
+	  if [[ $(echo $line | grep -e "^- \[.*\]\(.*\)$") ]]; then
+	    file=$(echo $line | cut -d'(' -f2 | cut -d')' -f1)
+	    file_path="$topic_doc_path/$file"
+	    echo "<!-- Extracted from $topic_doc_path/$file -->" >> "$expanded_topic_path"
+	    echo >> "$expanded_topic_path"
+	    cat "$file_path" >> "$expanded_topic_path"
+	    echo >> "$expanded_topic_path" # Ensure that there is always a newline between files.
+	  else
+	    echo "$line" >> "$expanded_topic_path"
+	  fi
+	done
+
+}
```

Comment:
This looks like it was extracted from https://github.com/material-components/material-components-ios/blob/628b2ecb9020b0287f44b384b93946977fd07b8a/scripts/generate_readme#L101-L114, please extract it from that script to a shared `.sh` file in a separate PR first, and then both scripts can rely on the method.

Reply:
Done